crates/pathfinder/src/sequencer.rs

```diff
@@ -821,43 +821,87 @@ mod tests {
     }
 
     mod state_update_by_number_matches_by_hash_on {
-        use super::*;
+        use super::{
+            reply::{
+                state_update::{Contract, StorageDiff},
+                StateUpdate,
+            },
+            *,
+        };
+        use crate::core::{ContractAddress, GlobalRoot};
         use pretty_assertions::assert_eq;
+        use std::collections::{BTreeSet, HashMap};
+
+        #[derive(Clone, Debug, PartialEq)]
+        pub struct OrderedStateDiff {
+            pub storage_diffs: HashMap<ContractAddress, BTreeSet<StorageDiff>>,
+            pub deployed_contracts: BTreeSet<Contract>,
+        }
+
+        #[derive(Clone, Debug, PartialEq)]
+        pub struct OrderedStateUpdate {
+            pub new_root: GlobalRoot,
+            pub old_root: GlobalRoot,
+            pub state_diff: OrderedStateDiff,
+        }
+
+        impl From<StateUpdate> for OrderedStateUpdate {
+            fn from(s: StateUpdate) -> Self {
+                Self {
+                    new_root: s.new_root,
+                    old_root: s.old_root,
+                    state_diff: OrderedStateDiff {
+                        storage_diffs: s
+                            .state_diff
+                            .storage_diffs
+                            .into_iter()
+                            .map(|(addr, diffs)| (addr, diffs.into_iter().collect()))
+                            .collect(),
+                        deployed_contracts: s.state_diff.deployed_contracts.into_iter().collect(),
+                    },
+                }
+            }
```

Comment:
This does make the assumption that there are no repeat storage updates for a contract. I presume there shouldn't be, but there can of course also be :). But should be fine since this is for a test.

Reply:
Luckily the very contracts in those two test cases don't suffer from this, and this is supposed to be just a smoke test.